src/util.c

```diff
@@ -645,8 +645,33 @@ printsocket(struct tcb *tcp, int fd, const char *path)
 	return true;
 }
 
+typedef bool (*scan_fdinfo_fn)(const char *value, void *data);
+
+static bool
+scan_fdinfo(pid_t pid_of_fd, int fd, const char *search_pfx,
+	    size_t search_pfx_len, scan_fdinfo_fn fn, void *data);
+
+#define is_ptmx(FINFO) ((FINFO)->type == FINFO_DEV_CHR \
+			&& (FINFO)->dev.major == 5 \
+			&& (FINFO)->dev.minor == 2)
+
+static bool get_pts(const char *value, void *data)
+{
+	struct finfo *finfo = data;
+
+	errno = 0;
+	long i = strtol(value, NULL, 10);
+	if (i < 0 || i > INT_MAX || errno) {
+		debug_func_perror_msg("converting \"%s\" to int", value);
+		return false;
+	}
+
+	finfo->dev.tty_index = i;
+	return true;
+}
```

Comment:
Can't we use something from src/string_to_uint.h here?

Reply:
Updated.